validator/client/runner.go

```diff
@@ -120,13 +109,8 @@ func run(ctx context.Context, v iface.Validator) {
 			performRoles(slotCtx, allRoles, v, slot, &wg, span)
 		case isHealthyAgain := <-healthTracker.HealthUpdates():
 			if isHealthyAgain {
-				headSlot, err = initializeValidatorAndGetHeadSlot(ctx, v)
-				if err != nil {
-					log.WithError(err).Error("Failed to re initialize validator and get head slot")
-					continue
-				}
-				if err := v.UpdateDuties(ctx, headSlot); err != nil {
-					handleAssignmentError(err, headSlot)
+				if err = v.Init(ctx); err != nil {
+					log.WithError(err).Error("Failed to re initialize validator")
```

Comment:
Should we redo everything? Maybe we should introduce a separate `ReInit` method or add a bool parameter to `Init`?
- `WaitForKeymanagerInitialization` is not needed again
- `WaitForActivation` is not needed again because we already have an active validator if we reach this code path.
- `CheckDoppelGanger` is maybe not needed again?

And we are also ignoring all errors here, similarly to when we call `Init` for the first time

Reply:
hmm that's a good point maybe we just need to update duties or wait for chain start